src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+      return element;
+    };
+
+    /**
+     * Destroys the video technology factory
+     * @public
+     * @method OoyalaVideoFactory#destroy
+     */
+    this.destroy = function() {
+      this.encodings = [];
+      this.create = function() {};
+    };
+
+    this.maxSupportedElements = -1;
+  };
+
+  /* 
+   *This function creates an <iframe> (YouTube player)
+   *after the API code downloads.
+   */
+  function onYouTubeIframeAPIReady() {
+    youtubePlayer = new YT.Player('player', {
+      videoId: youtubeID,
+      height: "100%",
+      width: "100%",
+      playerVars: { 'autoplay': 0, 'controls': 0 },
+      events: {
+      'onReady': onPlayerReady,
+      'onPlaybackQualityChange': onPlayerPlaybackQualityChange,
+      'onStateChange': onPlayerStateChange,
+      'onError': onPlayerError
+      }
+    });
+  };
+
+  /*
+   *The Youtube iframe API will call this function when the video player is ready.
+   */
+  function onPlayerReady(event) {
+    playerReady = true;
+    if (!playerNotready) {
+      for(var i = 0; i < javascriptCommandQueue.length; i++) 
+      {
+        if(javascriptCommandQueue[i][0] == "play")
+        { 
+          element.play();
+          hasPlayed = true;
+        }
+        else if(javascriptCommandQueue[i][0] == "seek")
+        {  
+          element.seek(javascriptCommandQueue[i][1])
+        }    
+      }
+    }
+  };
+
+  /*
+   *The Youtube iframe API calls this function when the player's quality changes.
+   */
+  function onPlayerPlaybackQualityChange(event) {
+    var bitrateValue = event.data;
+    var vtcBitrate = {
+      id: event.data,
+      width: 0,
+      height: 0,
+      bitrate:event.data 
+    }
+    element.controller.notify(element.controller.EVENTS.BITRATE_CHANGED,vtcBitrate);
+  };
+
+  /*
+   *The Youtube iframe API calls this function when the player's state changes.
+   */
+  function onPlayerStateChange(event) {
+    switch(event.data) {
+      case -1:
+       // unstarted
+        break;
+      case 0:
+        // ended
+        videoEnded = true;
+        element.controller.notify(element.controller.EVENTS.ENDED);
+        break;
+      case 1:
+        // playing 
+	OO.log("Youtube: Playing event received");
+        element.controller.notify(element.controller.EVENTS.PLAYING);
+
+        if (bitrateFlag) {
+          qualities = youtubePlayer.getAvailableQualityLevels();
+          element.raiseBitratesAvailable();
+          bitrateFlag = false;
+        }      
+        break;
+      case 2:
+        // paused 
+        element.controller.notify( element.controller.EVENTS.PAUSED);
+        break;
+      case 3:
+        // buffering
+        OO.log("Youtube: Buffering event received"); 
+        break;
+      case 5:
+        // video cued 
+        break;
+    }
+  };
+
+  /*
+   *The Youtube iframe API calls this function when the player's throws an error.
+   */
+  function onPlayerError(event) {
+    var code = -1;
+    switch(event.data) {
+      case 2:
```

Comment:
Are there constants we can reference instead of 2,5,100,etc or do we need to use these numbers?